content/tutorial-deep-learning-on-mnist.md

```diff
@@ -165,6 +171,9 @@ for sample, ax in zip(rng.choice(x_train, size=num_examples, replace=False), axe
     ax.imshow(sample.reshape(28, 28), cmap='gray')
 ```
 
+_Above are five images taken from the MNIST training set. Hand-drawn
+Arabic numerals 3, 8, 9, 6, and 5._
```

Comment:
This highlights an interesting concern related to alt-text for non-static images. The caption provides specific info about the image which will be correct as long as nothing changes in the code that generates the image. For example, if the user changes the seed then this caption will very likely no longer be correct. One way to avoid this would be to use tools to embed the computed values into the text, e.g. the `glue` functionality offered by myst-nb. Unfortunately we can't use that here because it's not (yet) supported in Jupyter notebooks.

So there's a fundamental conflict here - good alt-text (AIUI) aims to be as descriptive and specific as possible, but in the case of generated images, things like this run the risk of falling out of sync with the images themselves and becoming incorrect. I'm not sure what the right answer is for cases like these.

Reply:
I wasn't sure what the "right" thing to do here is either. I hadn't considered the effect from seeding or even just running a second time. 

I do think the error would decrease even for different seeds, but I didn't try. 

The deeper issue of generated image alt text, \_(:/)_/
Would love to get another opinion on best practices.

Reply:
I reached out to 2i2c to see if there is any standard practice for alt-text. 

@choldgraf suggested using [`glue`](https://myst-nb.readthedocs.io/en/latest/use/glue.html). 

Perhaps our alt-text could be saved as a glued variable `alt-text` as the figure is plotted. The drawback is that the Jupyter interface will have a markdown cell with ``{glue:} `alt-text` ``

Reply:
Yup that's what I was thinking too, but we're currently not using any myst-nb features that aren't supported in both `.html` and notebook outputs. If we were, then this would be a non-issue